UnityGLTF/Assets/UnityGLTF/Scripts/GLTFSceneImporter.cs

```diff
@@ -1281,6 +1290,8 @@ protected virtual async Task ConstructMesh(GLTFMesh mesh, Transform parent, int
 
 			for (int i = 0; i < mesh.Primitives.Count; ++i)
 			{
+				cancellationToken.ThrowIfCancellationRequested();
+
 				var primitive = mesh.Primitives[i];
 				int materialIndex = primitive.Material != null ? primitive.Material.Id : -1;
 
```

Comment:
For the two awaited calls below (ConstructMeshPrimitive and SetupBones), you should either call ThrowIfCancellationRequested after calling them, or pass the cancellationToken in to them and have them ThrowIfCancellationRequested after each of their awaits.